streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -60,11 +63,17 @@ public void makeReady(final Map<InternalTopicConfig, Integer> topics) {
                 final MetadataResponse metadata = streamsKafkaClient.fetchMetadata();
                 final Map<String, Integer> existingTopicPartitions = fetchExistingPartitionCountByTopic(metadata);
                 final Map<InternalTopicConfig, Integer> topicsToBeCreated = validateTopicPartitions(topics, existingTopicPartitions);
+                if (metadata.brokers().size() < replicationFactor) {
+                    throw new StreamsException("Not enough brokers " + metadata.brokers().size() +
```

Comment:
Nit: I think the error message does not read well:
`Not enough brokers 2 for replication factor 3`

Better: `Found only 2 brokers, but replication factor is 3. Decrease replication factor for internal topics via StreamsConfig parameter "replication.factor" or add brokers to your cluster.`